### DISCUSSION TITLE
search highlighting in pdf by modifying url

### DISCUSSION THREAD

**Discussion by conrad82** [jonaswinkler/paperless-ng#858] [Feature Requests]
Is it possible to add search highlighting when viewing the pdf's?

I use firefox and "PDF viewer in browser", and search highlighting is supported by adding "#search=searchterm" at the end of the url, e.g. https://paperless.example.com/api/documents/12/preview/#search=searchterm

I understand this may be different for different browser, so is a solution to append the view-url through settings?

In my use case, I could then specify in the settings: `Append view url: #search=%s` or something similar, where %s was the search term in paperless

Sorry if this has already been suggested, I could not find it when doing a quick search.

**Comment by shamoon** [comment]
Do you have a link to documentation of this feature? Didn't know it existed. Doesnt seem to work in Safari or Chrome, unfortunately, so perhaps limited use case but it is cool when it works (screenshot from FF)
<img width="801" alt="Screen Shot 2021-04-03 at 11 40 01 AM" src="https://user-images.githubusercontent.com/4887959/113488190-58fb1f80-9471-11eb-83b9-331fbf7c4a62.png">

**Reply by conrad82** [reply]
I searched duckduckgo and found it in the mozilla github issues 😬 I tried it, and it worked. 
https://github.com/mozilla/pdf.js/issues/1875 discusses the implementation for pdf.js in Firefox.

I searched for the Adobe PDF Open Parameters mentioned in the thread (links did not work), and found the one for Adobe Acrobat 7.0.5 (from 2005): https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/PDFOpenParameters.pdf
So I guess this is the standard / documentation.
I can also confirm that #search= works in adobe acrobat reader plugin for internet explorer. 

![image](https://user-images.githubusercontent.com/17353007/113488588-60491a00-94bf-11eb-9410-08c7890597ee.png)
![image](https://user-images.githubusercontent.com/17353007/113488621-8373c980-94bf-11eb-9f05-7d44faa0b534.png)

Since I do not how if and how this is supported in other browsers, i propose adding a custom url append in the settings.
Too bad if it is not supported by chrome / safari pdf readers.

**Reply by conrad82** [reply]
I did some more searching, apparently it is an open issue in chromium (since 2017), but it seems the implementation is tricky
https://bugs.chromium.org/p/chromium/issues/detail?id=792647

**Reply by jonaswinkler** [reply]
> internet explorer

=)

**Reply by conrad82** [reply]
> > internet explorer
> 
> =)

I tested it on my work computer! 🙈 (edit: it also has chrome)
On private PC, it is linux all the way, promise 😅

**Comment by shamoon** [comment]
I went ahead and implemented a PR fro this #860 because why not. It seems to have no negative effects in other browsers but is a cool 'progressive enhancement' for FF (and maybe someday others...). It takes the current search query on search results page or the text in the text field when filtering documents and uses that to highlight.

**Comment by iwishiwasaneagle** [comment]
Is it possible to implement a button to turn this off? I often search my archive for keywords and opening a document to read where there's a rainbow of highlighted searches on the page is very off-puting. Maybe, when there is a active search term, show a 2nd preview button?

![image](https://user-images.githubusercontent.com/12383029/143232260-723e6278-354f-4135-b884-a2d25b0572a4.png)
